File: data/code_pairs/fortran/123.f90
subroutine networkinum(ipkon, inum, kon, lakon, ne, itg, ntg, network)
  implicit none

  integer, intent(in) :: ipkon(*), kon(*), ne, itg(*), ntg, network
  integer, intent(inout) :: inum(*)
  character(len=8), intent(in) :: lakon(*)
  
  integer :: i, indexe, node1, node2, node3
  
  do i = 1, ntg
     if (inum(itg(i)) > 0) inum(itg(i)) = -inum(itg(i))
  end do

  do i = 1, ne
     if (ipkon(i) < 0) cycle
     if ((lakon(i)(1:1) /= 'D') .and. ((lakon(i)(1:1) /= 'D') .or. (network /= 1))) cycle
     
     indexe = ipkon(i)
     if (kon(indexe+1) /= 0) then
        node1 = kon(indexe+1)
        if (inum(node1) > 0) inum(node1) = -inum(node1)
     end if
     node2 = kon(indexe+2)
     if (inum(node2) > 0) inum(node2) = -inum(node2)
     if (kon(indexe+3) /= 0) then
        node3 = kon(indexe+3)
        if (inum(node3) > 0) inum(node3) = -inum(node3)
     end if
  end do
end subroutine

File: data/code_pairs/cpp/123.cpp
#include <vector>
#include <string>

void networkinum(std::vector<int>& ipkon, std::vector<int>& inum, std::vector<int>& kon, 
                 const std::vector<std::string>& lakon, int ne, std::vector<int>& itg, 
                 int ntg, int network) {
    
    for (int i = 0; i < ntg; i++) {
        if (inum[itg[i]-1] > 0) inum[itg[i]-1] = -inum[itg[i]-1];
    }

    for (int i = 0; i < ne; i++) {
        if (ipkon[i] < 0) continue;
        std::string lakonl = lakon[i];
        if ((lakonl[0] != 'D') && ((lakonl[0] != 'D') || (network != 1))) continue;

        int indexe = ipkon[i] - 1; // Adjust for 0-based indexing
        if (kon[indexe + 1] != 0) {
            int node1 = kon[indexe + 1] - 1; // Adjust for 0-based indexing
            if (inum[node1] > 0) inum[node1] = -inum[node1];
        }
        int node2 = kon[indexe + 2] - 1; // Adjust for 0-based indexing
        if (inum[node2] > 0) inum[node2] = -inum[node2];
        
        if (kon[indexe + 3] != 0) {
            int node3 = kon[indexe + 3] - 1; // Adjust for 0-based indexing
            if (inum[node3] > 0) inum[node3] = -inum[node3];
        }
    }
}